packages/library/src/commonMain/kotlin/io/realm/RealmConfiguration.kt

```diff
@@ -17,52 +17,54 @@
 package io.realm
 
 import io.realm.internal.Mediator
+import io.realm.internal.RealmModelInternal
+import io.realm.internal.RealmObjectCompanion
 import io.realm.interop.NativePointer
 import io.realm.interop.RealmInterop
 import io.realm.interop.SchemaMode
-import io.realm.interop.Table
+import kotlin.reflect.KClass
 
-class RealmConfiguration private constructor(
-    val path: String?, // Full path if we don't want to use the default location
-    val name: String?, // Optional Realm name (default is 'default')
-    val schema: Mediator,
-    val version: Long = 0,
-    val tables: List<Table> = listOf()
+class RealmConfiguration(
```

Comment:
As discussed. I think it would be a mistake to remove the builder as the builder has better discoverability and autocomplete in the IDE. Required parameters like schema do need to be constructor parameters though and it might make sense to have a shorthand to create common configurations directly, but not sure if having multiple ways to construct the configuration is too confusing?

But IMO having these two options, seems like a good tradeoff to me:

```
// Construct "common" configurations directly
RealmConfiguration(val name = "default.realm",  val schema: Set<KClass<out RealmObject>>)

// Anything else more complicated requires you to go through the builder
RealmConfiguration.Builder(val schema: Set<KClass<out RealmObject>>)
class Builder {
  fun name(...)
  fun directory(...)
  fun otherMethods(...)
}
```

Thoughts?


```

Reply:
I like the concept of having a simple option and a full blown one. I am a bit in favor of the `schema(...)` method instead of required arguments to the constructor, as that is more fluent when there are many classes involved. But I do get the that it is in fact required until we potentially could find a way to gather a default schema behind the scenes, so don't know if it is too loose.